src/main/java/lotto/model/WinningLotto.java

```diff
@@ -0,0 +1,50 @@
+package lotto.model;
+
+import java.util.Set;
+
+public class WinningLotto {
+    private static final String BONUS_BALL_CONTAIN_LOTTO_NUMBER_MESSAGE = "로또넘버와 보너스볼은 중첩 될 수 없습니다.";
+
+    private final Lotto winningLotto;
+    private final LottoNumber bonusBall;
+
+    public WinningLotto(Set<LottoNumber> lottoNumbers, LottoNumber bonusBall) {
+        validateBonusBallNotContainLottoNumbers(lottoNumbers, bonusBall);
+        this.winningLotto = new Lotto(lottoNumbers);
+        this.bonusBall = bonusBall;
+    }
+
+    private void validateBonusBallNotContainLottoNumbers(Set<LottoNumber> lottoNumbers, LottoNumber bonusBall) {
+        if (lottoNumbers.contains(bonusBall)) {
+            throw new IllegalArgumentException(BONUS_BALL_CONTAIN_LOTTO_NUMBER_MESSAGE);
+        }
+    }
+
+    public WinningState makeWinningState(Lotto lotto) {
+        int winningCount = makeWinningCount(lotto.getLottoNumbers(), getLottoNumbers());
+        WinningState winningState = new WinningState(winningCount, checkBonusCount(lotto, this.bonusBall));
+
+        return winningState;
+    }
+
+    private int makeWinningCount(Set<LottoNumber> lottoNumbers, Set<LottoNumber> winningLottoNumbers) {
+        return winningLottoNumbers.stream()
+                .mapToInt((winningLottoNumber) -> compareWithWinningLotto(lottoNumbers, winningLottoNumber))
+                .sum();
+    }
+
+    private int compareWithWinningLotto(Set<LottoNumber> lottoNumbers, LottoNumber winningLottoNumber) {
+        if (lottoNumbers.contains(winningLottoNumber)) {
+            return 1;
+        }
+        return 0;
+    }
```

Comment:
아래와 같이 변경해볼 수도 있을 것 같아요 😃 
```suggestion
        return (int) winningLottoNumbers.stream()
                .filter((winningLottoNumber) -> lottoNumbers.contains(winningLottoNumber))
                .count();
    }
```

Reply:
오 그러네요 👍
정답로또를 로또에 비교하기보단
로또를 정답로또에 비교하는게 좀 더 논리적인것 같아 그 부분만 수정했습니다! 😀